students/morgan/mailprog/mailprog/mailprog/dataModels.py

```diff
@@ -0,0 +1,181 @@
+import csv
+import copy
+
+
+
+
+
+
+
+
+
+class Donor(object):
+
+    def __init__(self, name , donations= None):
+        self.name = name
+        self.donations = donations
+        self.i = 0
+
+    def add_donation(self, money):
+        self.donations.append(money)
+
+    def __iter__(self):
+        for item in self.donations:
+            yield item
+
+    def __next__(self):
+        if self.i < len(self.donations):
+            self.i += 1
+            return self.donations[self.i]
+
+    @property
+    def total(self):
+        return sum(self.donations)
+
+    @property
+    def maxdonation(self):
+        return max(self.donations)
+
+    @property
+    def namelength(self):
+        return (len(self.name))
+
+    @property
+    def donation_count(self):
+        return (len(self.donations))
+
+    @property
+    def average_donation(self):
+        return (self.total / self.donation_count)
+
+
+class Storage:
+    def __init__(self, donors = []):
+        self.donors = donors
+        self.user = ""
+
+
+
+    def __iter__(self):
+        return self.donors
+
+    def list_donors(self):
+        temp = []
+        for x in self.donors:
+            temp.append(x.name)
+        return temp
+
+    def record_user(func):
```

Comment:
this should probably be called `_record_user`, so it's clear you don't want to call it outside this class.